test/unit/specs/hive/key-value.js

```diff
@@ -0,0 +1,254 @@
+        await expect(() => store.set(0, 'testValue')).to.throw(errorMsg)
+        await expect(() => store.set(true, 'testValue')).to.throw(errorMsg)
+        await expect(() => store.set(123, 'testValue')).to.throw(errorMsg)
+        await expect(() => store.set(() => undefined), 'testValue').to.throw(errorMsg)
+      })
+
+      it('fails when object is empty', async () => {
+        const errorMsg = 'Provided object must have at least 1 key.'
+
+        await expect(() => store.set({})).to.throw(errorMsg)
+      })
+
+      it('fails when options is invalid', async () => {
+        const errorMsg1 = 'Expiration seconds must be a number.'
+
+        await expect(() => store.set('k', 'v', { expirationSeconds: null })).to.throw(errorMsg1)
+        await expect(() => store.set('k', 'v', { expirationSeconds: false })).to.throw(errorMsg1)
+        await expect(() => store.set('k', 'v', { expirationSeconds: true })).to.throw(errorMsg1)
+        await expect(() => store.set('k', 'v', { expirationSeconds: '' })).to.throw(errorMsg1)
+        await expect(() => store.set('k', 'v', { expirationSeconds: 'foo' })).to.throw(errorMsg1)
+        await expect(() => store.set('k', 'v', { expirationSeconds: NaN })).to.throw(errorMsg1)
+        await expect(() => store.set('k', 'v', { expirationSeconds: () => undefined })).to.throw(errorMsg1)
+
+        const errorMsg2 = 'Expiration must be one of this values: TTL, UnixTimestamp, None.'
+
+        await expect(() => store.set('k', 'v', { expiration: null })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: false })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: true })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: '' })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: 'foo' })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: NaN })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: () => undefined })).to.throw(errorMsg2)
+
+        const errorMsg3 = 'Condition must be one of this values: IfExists, IfNotExists, Always.'
+
+        await expect(() => store.set('k', 'v', { condition: null })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: false })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: true })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: '' })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: 'foo' })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: NaN })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: () => undefined })).to.throw(errorMsg3)
+      })
+    })
+
+    describe('Increment', () => {
+      it('success', async () => {
+        store = Backendless.Hive(hiveName).KeyValueStore('storeKey')
```

Comment:
move it to `beforeEach`

Reply:
see https://github.com/Backendless/JS-SDK/pull/228#discussion_r883417954